examples/shouldi/shouldi/cli.py

```diff
@@ -182,12 +183,67 @@ async def is_lang_javascript(self, repo):
                 return {"javascript": True}
 
 
+DATAFLOW_SA_JAVASCRIPT = DataFlow.auto(
+    pypi_package_json,
+    pypi_latest_package_version,
+    pypi_package_url,
+    pypi_package_contents,
+    cleanup_pypi_package,
+    run_npm_audit,
+    GetSingle,
+)
+
+DATAFLOW_SA_JAVASCRIPT.seed.append(
+    Input(
+        value=[run_npm_audit.op.outputs["report"].name,],
+        definition=GetSingle.op.inputs["spec"],
+    )
+)
+
+
+@op(
+    inputs={"repo": clone_git_repo.op.outputs["repo"]},
+    outputs={"result": SA_RESULTS},
+    conditions=[is_lang_javascript.op.outputs["javascript"]],
+)
+async def run_javascript_sa(self, repo):
+    """
+    Run JS static analysis
+    """
+    setup_kwargs = SetupPyKWArg.get_kwargs(
+        os.path.join(repo.directory, "package.json")
+    )
+
+    # TODO Make is so that
+    async with self.octx.parent(DATAFLOW_SA_JAVASCRIPT) as octx:
+        async for _, results in octx.run(
+            [
+                Input(
+                    value=setup_kwargs["name"],
+                    definition=pypi_package_json.op.inputs["package"],
```

Comment:
Here you'll want to have the input be the input for `run_npm_audit`, so you'll want to make the definition, `run_npm_audit.op.inputs["pkg"]`, the value would be `repo.directory`. So this gives the dataflow an input which it sees should go to `run_npm_audit` which is going to use that as it's `pkg` argument.